.dockerhub/debug/hooks/build

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+FROM_IMAGE=osrf/ros2:nightly
+FAIL_ON_BUILD_FAILURE=""
+UNDERLAY_MIXINS="debug ccache"
+OVERLAY_MIXINS="debug ccache coverage-gcc"
```

Comment:
This mixin might need some additional work before we can use it. I can't remember the exact problem, but I needed to add the --coverage option to the linker commands as well. 
https://github.com/ros-planning/navigation2/blob/59f0360827a504ac9d54dfbc7019cd8547fc766c/nav2_common/cmake/nav2_package.cmake#L42-L43

I don't remember the exact problem though. I feel like it just left out coverage data rather than giving a linker error.

Reply:
Please review this related PR:
https://github.com/colcon/colcon-mixin-repository/pull/20

Reply:
@crdelsey , looks like the above PR has merged.

Reply:
I just confirmed the behavior. If I don't add the linker flags, some files just silently don't report coverage data. `install/include/costmap_queue\costmap_queue.hpp` is a good example. If I use the mixin, this file reports 0% coverage. If I add the linker flags, I get 67% coverage.

I don't fully understand why this is happening, though.

Reply:
I think the PR is good. I believe using --coverage allows compatibility with Clang. I'm not sure if it fixes this problem though. I'll try to do some tests today to figure out what's going on.

Reply:
Make sure you `colcon mixin update <name>` for the name you used when adding the default mixin-repository linked above, so that you have the latest changes.

Reply:
Yeah. It looks good now. I wonder why we didn't get linking errors before. But, I'm not curious enough to bother testing; I'm just happy that this works :-)